### DISCUSSION TITLE
Mastodon

### DISCUSSION THREAD

**Discussion by kimsvane** [tteck/Proxmox#700] [Ideas]
Hi guys

Could you make a script for installing Mastodon in LXC ? 
https://docs.joinmastodon.org/

**Comment by Vakrehus** [comment]
Is there a chance to get a script for **Mastodon** in the future?
Tried several times to install it in LXC under Debian 11, with own PostgreSQL. It's just terrible and I can't get it to run.
Does anyone have a running LXC ?

**Comment by kimsvane** [comment]
I Got it installer on a lxc, so it’s possible. But I retried several times before it got installed. There is no lxc script, but I followed the guides onlineVenligst Kim Søndergaard Svane On 7 Jan 2023, at 19.09, Vakrehus ***@***.***> wrote:﻿
Is there a chance to get a script for Mastodon in the future?
Tried several times to install it in LXC under Debian 11, with own PostgreSQL. It's just terrible and I can't get it to run.
Does anyone have a running LXC ?

**Comment by Vakrehus** [comment]
The installation is not the problem. I rather have the problems to see the GUI of mastodon.
Mastodon in a LXC behind a Nginx Proxy Manager (LXC) behind cloudflare and to set up the nginx-config of the Mastodon container cleanly is rather my problem.

Followed this steps, used my own PostgreSQL database (LXC) and converted the database template to UTF8 (default is SQL_:ASCII - not working with mastodon)

UTF8-convert: https://www.shubhamdipt.com/blog/how-to-change-postgresql-database-encoding-to-utf8/

Guide for Mastodon: https://wiki.vikings.net/Installing_Mastodon_from_Source_on_Debian_11_(Bullseye)

**Comment by kimsvane** [comment]
The exactly same way I did set it up. I had issues with elements not showing, file rights and permission was the issue first. Then I recall issues with proxy, though I don’t remember what solved it here. Venligst Kim Søndergaard Svane On 8 Jan 2023, at 11.18, Vakrehus ***@***.***> wrote:﻿
The installation is not the problem. I rather have the problems to see the GUI of mastodon.
Mastodon in a LXC behind a Nginx Proxy Manager (LXC) behind cloudflare and to set up the nginx-config of the Mastodon container cleanly is rather my problem.
Followed this steps, used my own PostgreSQL database (LXC) and converted the database template to UTF8 (default is SQL_:ASCII - not working with mastodon)
UTF8-convert: https://www.shubhamdipt.com/blog/how-to-change-postgresql-database-encoding-to-utf8/
Guide for Mastodon: https://wiki.vikings.net/Installing_Mastodon_from_Source_on_Debian_11_(Bullseye)

**Comment by ScubyG** [comment]
I have attempted to install a Mastodon server within a Debian 11 LXC (tteck script) following the official documentation at:
https://docs.joinmastodon.org/admin/install

I tried the installation on Debian 12, all the Ubuntu's and Alpine Linux without success.

One of the issue I encounted was the section on installing Node.js:
`curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg`

The path `/etc/apt/keyrings/` did not exist so I manually created the folder path with `mkdir /etc/apt/keyrings/` which the comand worked.

Another issue I had was to do with the locale of the LXC. I changed the default to 'UTF-8' or 'en_GB UTF-8' as I am in the UK. Postgres should default to UTF-8 which is required for Mastodon to work and install correctly, as previously mentioned by @Vakrehus there is a way to convert the Postgres database to UTF-8, however this would not work for me, or maybe I did something wrong whilst following that guide. After I tried to reinstall the Mastodon server this was still an issue and using `--trace` after the setup command stated there were issues with UTF-8 not being compatible with SQL-ASCII.

The other issue I still have and can not resolve (maybe it is linked to the UTF-8/SQL-ASCII issue?) is the setup command: `RAILS_ENV=production bundle exec rake mastodon:setup` of the Mastodon server, I still receive a whole bunch or errors, or what appear to be errors:

``/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railtie
s-7.0.8/lib/rails/command.rb:51:in `invoke'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railtie
s-7.0.8/lib/rails/commands.rb:18:in `<main>'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/bootsna
p-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_req
uire.rb:32:in `require'
Tasks: TOP => db:setup => db:create
(See full trace by running task with --trace)
That failed! Perhaps your configuration is not right
``

I am determined to get this to work so off for attempt 101...

**Reply by ScubyG** [reply]
Update...

I have a working Mastodon server running on a Debian 11 LXC, im going to smoke test it and see if it is stable, if so I will create a new one using Debian 12 or Alpine Linux if it allows and will look at contributing a guide on setting up a Mastodon server from start to finish from tteck lxc script and with nginx reverse proxy via Cloudflare.
